src/google/protobuf/stubs/port.h

```diff
@@ -252,9 +252,13 @@ static const uint64 kuint64max = GOOGLE_ULONGLONG(0xFFFFFFFFFFFFFFFF);
 #define GOOGLE_GUARDED_BY(x)
 #define GOOGLE_ATTRIBUTE_COLD
 
+#ifndef GOOGLE_PROTOBUF_USE_UNALIGNED
```

Comment:
It lets you put -DGOOGLE_PROTOBUF_USE_UNALIGNED=0 on the command line to disable this so ubsan doesn't complain on __x86_64__ (that's the only platform I have ubsan working usefully on to test with; applies to the others too).